### PR TITLE
Update non-GCE Salt distros to 1.6.0, fallback to ContainerVM Docker version on GCE

### PR DESCRIPTION
**Pull request by zmerlynn**
This is neutral on the current version of ContainerVM, and saves us some cycles.

### CODE REVIEW

**Comment by zmerlynn** [comment]
Wait, misread this Salt. Argh. We need this for GCE for the next ContainerVM, but I need special case the others.

**Comment by roberthbailey** [comment]
I changed the PR title to "WIP". Please flip it back once it's ready for review.

**Comment by zmerlynn** [comment]
Ready now. Tested the override path by flipping the the cloud provider equality to verify (I have a ContainerVM image with Docker 1.6 on it, so it resulted in a downgrade).

**Comment by zmerlynn** [comment]
cc @vmarmol @dchen1107

File: cluster/saltbase/salt/docker/init.sls [hunk 1]
```diff
@@ -60,8 +49,12 @@ purge-old-docker:
     - mode: 644
     - makedirs: true
 
-# We are caching the Docker deb file in GCS for reliability and speed.  To
-# update this to a new version of docker, do the following:
+# Docker is on the ContainerVM image by default. The following
+# variables are provided for other cloud providers, and for testing testing and dire circumstances, to allow
```
Comment (vmarmol) [inline cluster/saltbase/salt/docker/init.sls]: nit: double testing

**Comment by vmarmol** [comment]
LGTM, thanks @zmerlynn! Are we planning on updating the Docker version to 1.6 sometime?

**Comment by zmerlynn** [comment]
It's in the ContainerVM update. Do you want me to just bonk it into here since we've already e2e'd it? It would basically update Azure and AWS but not us (yet).

**Comment by vmarmol** [comment]
Yeah I was thinking that for the non-GCE setups. I'm fine doing it here or in another PR.

**Comment by zmerlynn** [comment]
I can do it here. One sec.

**Comment by zmerlynn** [comment]
Upgrading non-GCE providers to 1.6.0. Tested by briefly flipped the equality again and doing an `-up` on a stock ContainerVM image, then verifying the versions.

**Comment by zmerlynn** [comment]
PTAL (fixed the nit along the way)

**Comment by vmarmol** [comment]
LGTM, will merge on green. Thanks again @zmerlynn!

**Pull request merged**